csharp_dotnetcore/2.EchoBot-With-Counter/Startup.cs

```diff
@@ -48,7 +50,10 @@ public void ConfigureServices(IServiceCollection services)
         {
             services.AddBot<EchoWithCounterBot>(options =>
             {
-                options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+                // Load the connected services from .bot file.
+                var botConfig = BotConfiguration.Load(@".\EchoBotWithCounter.bot");
+                var endpointService = (EndpointService)botConfig.Services.First(s => s.Type == "endpoint");
```

Comment:
Why is a cast to EndpointService needed? Can we update the code so this isn't needed?